### PR TITLE
add debug operator for observables

### PR DESCRIPTION
**Pull request by sashei**
Fixes nothing, just something I've been wishing I had for debugging

- ~~add “WIP” to the PR title if pushing up but not complete nor ready for review~~
- [x] ~~double check the original issue to confirm it is fully satisfied~~
- [x] add testing notes and screenshots in PR description to help guide reviewers
- [x] add unit tests
  - optional: consider adding instrumentation (integration/UI) tests
- consider running this branch in a debug simulator and check for memory leak notifications or any warnings
- [x] request the "UX" team perform a design review (if/when applicable)
- [x] make sure CI builds are passing (e.g.: fix lint and other errors)

### CODE REVIEW

File: app/src/test/java/mozilla/lockbox/ObservableExtensionTest.kt [hunk 1]
```diff
@@ -21,7 +32,17 @@ private enum class InvalidAction {
...
+@RunWith(PowerMockRunner::class)
+@PrepareForTest(LogProvider::class)
 class ObservableExtensionTest {
+    @Mock
+    val logger = Mockito.mock(Logger::class.java)
+
+    @Before
+    fun setUp() {
+        PowerMockito.mockStatic(LogProvider::class.java)
+        Whitebox.setInternalState(LogProvider::class.java, "log", logger)
```
Comment (jhugman) [inline app/src/test/java/mozilla/lockbox/ObservableExtensionTest.kt]: Woah.

File: app/src/main/java/mozilla/lockbox/extensions/Observable+.kt [hunk 2]
```diff
@@ -7,13 +7,30 @@
...
 import io.reactivex.Observable
+import mozilla.lockbox.LogProvider
 import mozilla.lockbox.model.ItemViewModel
 import org.mozilla.sync15.logins.ServerPassword
 
 fun <T : Any, U : T> Observable<T>.filterByType(clazz: Class<out U>): Observable<U> {
     return this.filter { t -> clazz.isInstance(t) }.map { t -> clazz.cast(t) }
 }
 
+fun <T : Any> Observable<T>.debug(): Observable<T> {
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/extensions/Observable+.kt]: Nit: `log()` `toConsole()`, `logIt()` surely?

I'm not 1000% against `debug()` however. I just like my bikesheds brown.

File: app/src/test/java/mozilla/lockbox/ObservableExtensionTest.kt [hunk 3]
```diff
@@ -37,4 +58,23 @@ class ObservableExtensionTest {
 
         subscription.dispose()
     }
+
+    @Test
+    fun debug() {
```
Comment (jhugman) [inline app/src/test/java/mozilla/lockbox/ObservableExtensionTest.kt]: Nit: better name please.

File: app/src/main/java/mozilla/lockbox/extensions/Observable+.kt (hunk 2 above)
Comment (sashei) [reply app/src/main/java/mozilla/lockbox/extensions/Observable+.kt]: I'm just copying directly what was in RxSwift :pp

Comment (sashei) [reply app/src/main/java/mozilla/lockbox/extensions/Observable+.kt]: I'm just copying directly what was in RxSwift :pp

**Pull request merged**